Source/SIMPLib/REST/PipelineListener.h

```diff
@@ -18,28 +25,33 @@ class SIMPLib_EXPORT PipelineListener : public QObject, public IObserver
   PipelineListener(QObject* parent);
   virtual ~PipelineListener();
 
+  friend PipelineListenerMessageHandler;
+
   void createErrorLogFile(QString path);
   void createWarningLogFile(QString path);
   void createStatusLogFile(QString path);
-  void createStandardOutputLogFile(QString path);
   void closeFiles();
 
-  std::vector<PipelineMessage> getMessages();
-  std::vector<PipelineMessage> getErrorMessages();
-  std::vector<PipelineMessage> getWarningMessages();
-  std::vector<PipelineMessage> getStatusMessages();
-  std::vector<PipelineMessage> getStandardOutputMessages();
+  std::vector<const AbstractMessage*> getMessages();
+  std::vector<const AbstractErrorMessage*> getErrorMessages();
+  std::vector<const AbstractWarningMessage*> getWarningMessages();
+  std::vector<const AbstractStatusMessage*> getStatusMessages();
+  std::vector<const AbstractProgressMessage*> getProgressMessages();
 
   QString getErrorLog();
   QString getWarningLog();
   QString getStatusLog();
-  QString getStandardOutputLog();
 
 public slots:
-  void processPipelineMessage(const PipelineMessage& pm);
+  void processPipelineMessage(const AbstractMessage::Pointer& pm);
 
 private:
-  std::vector<PipelineMessage> m_Messages;
+  std::vector<AbstractMessage::Pointer> m_SharedMessages;
+  std::vector<const AbstractMessage*> m_Messages;
```

Comment:
Every other variable holding a collection of messages has something describing why the collection matters and what it is for.  In that context, m_Messages sounds like a variable that someone forgot to delete.  Consider renaming to m_AllMessages, or if that isn't accurate, some way of describing why it exists.  Consider updating the corresponding method to match.  Simply leaving it at Messages is too generic with respect to everything else and can easily be misused.   For instance, someone can parse through getMessages() and look for ErrorMessages because if someone is looking for messages, there is nothing in the name signifying there might be other collections already sorted.